File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.swift
import Foundation

func solution(_ n:Int, _ arr1:[Int], _ arr2:[Int]) -> [String] {
    var answer: [String] = []
    
    let binaryArr = zip(arr1, arr2).map { String($0.0 | $0.1, radix: 2) }
    
    let result = binaryArr.map {
        return n == $0.count ? $0 : Array(repeating: "0", count: n - $0.count) + $0
    }
    
    result.forEach { str in
        answer.append(str.map { ch in ch == "1" ? "#" : " " }.joined())
    }
    
    return answer
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <algorithm>
#include <iostream>
#include <map>
#include <string>
#include <vector>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;

    for (int i = 0; i < n; i++) {
        int num = arr1[i] | arr2[i];
        string s = "";

        while (num >= 1) {
            s += to_string(num % 2);
            num = num / 2;
        }

        while (s.size() != n) {
            s += "0";
        }

        reverse(s.begin(), s.end());

        answer.push_back(s);
    }

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            answer[i][j] == '1' ? answer[i][j] = '#' : answer[i][j] = ' ';
        }
    }

    return answer;
}